src/main/java/com/faforever/client/game/PlayerCardTooltipController.java

```diff
@@ -48,9 +53,44 @@ public Node getRoot() {
 
   @Override
   public void initialize() {
+    factionImage.managedProperty().bind(factionImage.visibleProperty());
+    factionIcon.managedProperty().bind(factionIcon.visibleProperty());
     foeIconText.managedProperty().bind(foeIconText.visibleProperty());
     foeIconText.setTooltip(new Tooltip(i18n.get("userInfo.foe")));
     friendIconText.managedProperty().bind(friendIconText.visibleProperty());
     friendIconText.setTooltip(new Tooltip(i18n.get("userInfo.friend")));
   }
+
+  private void setFactionIcon(Faction faction) {
+    if (faction == null) {
+      return;
+    }
+
+    switch (faction) {
+      case AEON:
+        factionIcon.setVisible(true);
+        factionIcon.setText("\uE900");
+        factionIcon.getStyleClass().add("aeon");
+        break;
+      case CYBRAN:
+        factionIcon.setVisible(true);
+        factionIcon.setText("\uE902");
+        factionIcon.getStyleClass().add("cybran");
+        break;
+      case SERAPHIM:
+        factionIcon.setVisible(true);
+        factionIcon.setText("\uE903");
+        factionIcon.getStyleClass().add("seraphim");
+        break;
+      case UEF:
+        factionIcon.setVisible(true);
+        factionIcon.setText("\uE904");
+        factionIcon.getStyleClass().add("uef");
+        break;
+      default:
+        factionImage.setVisible(true);
+        factionImage.setImage(randomImage);
+        break;
```

Comment:
Alex asked me about this earlier. I think in normal FAF having faction 5 means random, but I'm not sure about nomads. I would recommend you fire up a nomads game and see what the faction numbers for random and nomads are. It could be that in nomads 5 is nomads and 6 is random.

Reply:
Well i think what he does making every unknow display as random seems ok to me